config/hotreloadable.go

```diff
@@ -1,14 +1,52 @@
 package config
 
-import "time"
+import (
+	"sync/atomic"
+	"time"
+)
 
 // RegisterIntConfigVariable registers int config variable
+// Deprecated: use RegisterIntVar or RegisterAtomicIntVar instead
```

Comment:
It's good to have(if we aren't removing this func altogether)👍 
you may also use `registers a not hot-reloadable int config variable` like below to nudge them to use the newer and safer hot-reload registration.

Reply:
I also think it's good to have. I'd like to keep the old func to avoid breaking changes for other teams so that everybody has the time (and choice) to migrate as they see fit.

Reply:
But wouldn't keeping it defeat the purpose? We'd still be allowing race conditions right?

Reply:
Yeah I know, it's just to give people more time. I know it's not that big of a change, I think any IDE can do it very quickly so I'm good either way, just trying not to break things for others. Or we can do it incrementally like deprecate it first, after X months remove it. If someone pulls the new version then at that point they'll have to update it.